app/helpers/language_helper.py

```diff
@@ -30,6 +31,11 @@ def handle_language(metadata: Optional[MetadataProxy] = None) -> None:
         g.allowed_languages = get_allowed_languages(schema_name, launch_language)
         request_language = request.args.get("language_code")
         if request_language and request_language in g.allowed_languages:
+            if cookie_session.get("survey_title"):
```

Comment:
When would we not have a survey title at this point?. I would just check that we have metadata then proceed.